test/Microsoft.IdentityModel.Tests/IdentityComparer.cs

```diff
@@ -655,42 +656,60 @@ public static bool AreSecurityKeysEqual(SecurityKey securityKey1, SecurityKey se
             if (rsaKey1 != null && rsaKey2 != null)
             {
                 CompareAllPublicProperties(rsaKey1, rsaKey2, localContext);
-                AreRsaParametersEqual(rsaKey1.Parameters, rsaKey2.Parameters, localContext);
             }
 
             return context.Merge(localContext);
         }
 
-        public static bool AreRsaParametersEqual(RSAParameters rsaParameters1, RSAParameters rsaParameters2, CompareContext context)
+        public static bool AreRsaParametersEqual(object object1, object object2, CompareContext context)
         {
+            RSAParameters rsaParameters1 = (RSAParameters) object1;
+            RSAParameters rsaParameters2 = (RSAParameters) object2;
+
             var localContext = new CompareContext(context);
             if (!ContinueCheckingEquality(rsaParameters1, rsaParameters2, localContext))
                 return context.Merge(localContext);
 
             if (!AreBytesEqual(rsaParameters1.D, rsaParameters2.D, context))
+            {
+                localContext.Diffs.Add("D:");
                 localContext.Diffs.Add("!AreBytesEqual(rsaParameters1.D, rsaParameters2.D)");
-
+            }
```

Comment:
nit: normally we add a line after a closing brace.

Reply:
Added in the spaces.